src/xform_to_datalog/arcs_manifest_tree/handle_connection_spec.h

```diff
@@ -34,18 +36,45 @@ class HandleConnectionSpec {
         : (reads_) ? arcs::HandleConnectionSpecProto_Direction_READS
                    : arcs::HandleConnectionSpecProto_Direction_WRITES);
     result.set_name(name_);
+    *result.mutable_type() = type_->MakeProto();
     return result;
   }
 
+  // Get all of the AccessPaths rooted at the associated ParticleSpec
+  // (indicated through the provided name) through leaf fields of type_.
+  std::vector<raksha::ir::AccessPath> GetAccessPaths(
+      std::string particle_spec_name) const {
+    namespace ir = raksha::ir;
+    std::vector<ir::AccessPath> result_paths;
+    ir::AccessPathSelectorsSet selectors_set =
+        type_->GetAccessPathSelectorsSet();
+    ir::AccessPathRoot root(
+        ir::HandleConnectionSpecAccessPathRoot(
+            std::move(particle_spec_name), name_));
+    for (ir::AccessPathSelectors selectors :
+         ir::AccessPathSelectorsSet::CreateAbslSet(selectors_set)) {
+      result_paths.push_back(ir::AccessPath(root, std::move(selectors)));
+    }
+    return result_paths;
```

Comment:
I recall seeing similar code snippets elsewhere. It looks like there is an opportunity for refactoring this into a helper method. (Not necessary for this PR, though.)

Reply:
Agreed, I'll be looking for cleanup opportunities after MVP.